hass_nabucasa/cloud_api.py

```diff
@@ -25,17 +26,19 @@ def _log_response(func):
     @wraps(func)
     async def log_response(*args):
         """Log response if it's bad."""
-        resp = await func(*args)
-        meth = _LOGGER.debug if resp.status < 400 else _LOGGER.warning
-        meth("Fetched %s (%s)", resp.url, resp.status)
+        resp: ClientResponse = await func(*args)
+        if resp.ok:
+            _LOGGER.debug("Fetched %s (%d)", resp.url, resp.status)
+        else:
+            _LOGGER.warning("Fetched %s (%d): %s", resp.url, resp.status, resp.reason)
```

Comment:
The body contains the actual error message. If it's a 400, we should be safe to read the content and print that too.

Reply:
Is that not done by aiohttp? I saw on the HTTP parser, he set the reason flag on the response object

Reply:
`reason` is just the text version of the status code (400). So in this case that would be "Bad Request". It is not parsing the body and reading the JSON.

Reply:
Right, I was meaning that https://github.com/NabuCasa/hass-cloud/blob/master/utils/aws.py#L55 overwrite the default message we send on 400 and aiohttp parse that response based on the HTTP specification back to reason.